app/src/app/components/contents/admin/ManageRoles.tsx

```diff
@@ -10,10 +10,14 @@ import { setupManageRolesColumns } from "./utils/manageRolesColumns";
 import { rolesGlobalFilterFn } from "./utils/rolesTableGlobalFilterFn";
 
 import { NavLink } from "react-router-dom";
+import {Unauthorized} from "../common/Unauthorized";
 
 export const ManageRoles = () => {
   const { roles, users, mutate } = useManageAccessLayoutContext();
   const [filterValue, setFilterValue] = useState("");
+  if (!roles) {
+      return <Unauthorized />
+  }
```

Comment:
This, and corresponding addition in ManageUsers, is here because the parent AdminOutlet is now potentially accessible to users who have `packet.manage` permission (and can resync packets) but don't have `user.manage` permission. In this case the AdminOutlet does not attempt to get users and roles from the api. It's possible that such a user might manually navigate to `manage-roles` or `manage-users` in which case they will now see the Unauthorized screen rather than a blank page.